__tests__/app.test.js

```diff
@@ -72,79 +69,165 @@ describe("app", () => {
+            if (body.comments.length === 0) {
+              expect(body.comments).toEqual([]);
+            } else {
+              expect(body.comments).toBeSortedBy("created_at", {
+                descending: true,
+              });
+              body.comments.forEach((comment) => {
+                expect(typeof comment.comment_id).toBe("number");
+                expect(typeof comment.votes).toBe("number");
+                expect(typeof comment.created_at).toBe("string");
+                expect(typeof comment.author).toBe("string");
+                expect(typeof comment.body).toBe("string");
+                expect(typeof comment.article_id).toBe("number");
+              });
+            }
+          });
+      });
       test("GET 404: should return with an error if a valid article_id is given but it doesn't exist", () => {
         return request(app)
-        .get("/api/articles/100/comments")
-        .expect(404)
-        .then(({body}) => {
-          expect(body.msg).toBe("Not Found")
-        })    
-      })
+          .get("/api/articles/100/comments")
+          .expect(404)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Not Found");
+          });
+      });
       test("GET 400: should return with an error if an invalid file path is given", () => {
         return request(app)
-        .get("/api/articles/invalid/comments")
-        .expect(400)
-        .then(({body}) => {
-          expect(body.msg).toBe("Bad Request")
-        })
+          .get("/api/articles/invalid/comments")
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+    });
+  });
+  describe("POST test", () => {
+    describe("/api/articles/:article_id/comments", () => {
+      test("POST 200: should accept a valid comment post request to a specific article (i.e. article_id). It should respond with the posted comment", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            body: "excellent work",
+            username: "butter_bridge",
+          })
+          .expect(201)
+          .then(({ body }) => {
+            expect(body.addedComment).toEqual({
+              article_id: 1,
+              body: "excellent work",
+              votes: 0,
+              author: "butter_bridge",
+              created_at: String(new Date(Date.now())),
+              comment_id: 19,
+            });
+          });
+      });
+      test("POST 400: should respond with an error if there are missing keys in the post", () => {
+        return request(app)
+          .post("/api/articles/1/comments")
+          .send({
+            username: "butter_bridge",
+          })
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request");
+          });
+      });
+      test("POST 400: should return with an error if an invalid username is given", () => {
```

Comment:
I would suggest that an invalid username should be a 404 in this case, as it's a potentially valid username but just isn't found on the users table.